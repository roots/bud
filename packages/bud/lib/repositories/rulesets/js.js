"use strict";
exports.__esModule = true;
exports.js = void 0;
var js = function (bud) {
    return bud.hooks.filter('webpack.module.rules.js', {
        test: bud.hooks.filter('webpack.module.rules.js.test', bud.patterns.get('js')),
        exclude: bud.hooks.filter('webpack.module.rules.js.exclude', bud.patterns.get('vendor')),
        use: bud.hooks.filter('webpack.module.rules.js.use', {
            value: [bud.uses.get('babel')(bud)],
            bud: bud
        }).value
    });
};
exports.js = js;
//# sourceMappingURL=js.js.map