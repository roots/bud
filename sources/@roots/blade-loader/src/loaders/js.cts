export default function (source: string) {
  return [...source.matchAll(/@js(?<content>[\s\S]*?)@endjs/g)]
    .map(match => match.groups.content)
    .filter(Boolean)
    .join(`\n`)
}
