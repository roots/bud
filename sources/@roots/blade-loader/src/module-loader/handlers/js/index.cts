import {extract} from '@roots/blade-loader/module-loader/handlers'

export const extension = `js`
export const pattern =
  /@module\(['"]js['"]\)(?<content>[\s\S]*?)@endmodule/g

export default (source: string) => extract(source, pattern)
