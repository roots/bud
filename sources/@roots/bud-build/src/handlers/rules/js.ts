import type {Factory} from '../index.js'

export const js: Factory = ({filter, makeRule, path}) =>
  makeRule()
    .setTest(filter(`pattern.js`))
    .setInclude([() => path(`@src`)])
    .setUse([])
