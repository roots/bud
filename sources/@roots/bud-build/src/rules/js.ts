import type {Factory} from '@roots/bud-build/registry'

const js: Factory = async ({filter, makeRule, path}) =>
  makeRule()
    .setTest(filter(`pattern.js`))
    .setInclude([() => path(`@src`)])
    .setResolve({
      fullySpecified: false,
    })

export {js as default}
