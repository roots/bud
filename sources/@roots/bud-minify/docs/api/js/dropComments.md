---
title: bud.minify.js.dropComments
---

Remove comments from output:

```typescript
export default async bud => {
  bud.minify.js.dropComments()
}
```
