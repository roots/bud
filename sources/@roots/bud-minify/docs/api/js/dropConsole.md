---
title: bud.minify.js.dropConsole
---

Remove `console.*` statements from output.

```typescript
export default async bud => {
  bud.minify.js.dropConsole()
}
```
