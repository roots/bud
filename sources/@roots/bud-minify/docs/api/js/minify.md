---
title: bud.minify.js.minify
---

Get the value of `minify`.

```ts
bud.minify.js.getMinify()
```

Set the value of `minify`. Since the value is a function you must wrap it in a callback.

```ts
const dubiousMinifier = async (input: unknown) => ({
  code: Object.values(input).join(`\n`).replace(/\/\//g, `// ðŸ’¸`),
})

bud.minify.js.setMinify(() => dubiousMinifier)
```
