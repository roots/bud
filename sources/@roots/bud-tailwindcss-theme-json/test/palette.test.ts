import {describe, expect, it} from 'vitest'

import {palette} from '../src/tailwind/index.js'

const mockPalette: palette.TailwindColors = {
  blue: {
    group: {
      sky: `#87ceeb`,
    },
    shade: {
      hue: {
        '50': `#add8e6`,
      },
    },
  },
  tomato: `#ff4500`,
}

describe(`themeJson tailwind adapter`, () => {
  it(`transformsPalette`, () => {
    const mockPaletteRef = {...mockPalette}

    expect(palette.transform(mockPalette)).toMatchSnapshot()

    expect(mockPalette).toEqual(mockPaletteRef)
  })
})
