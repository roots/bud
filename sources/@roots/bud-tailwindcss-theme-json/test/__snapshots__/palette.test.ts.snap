// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`themeJson tailwind adapter > transformsPalette 1`] = `
[
  {
    "color": "#87ceeb",
    "name": "Blue Group Sky",
    "slug": "blue-group-sky",
  },
  {
    "color": "#add8e6",
    "name": "Blue Shade Hue 50",
    "slug": "blue-shade-hue-50",
  },
  {
    "color": "#ff4500",
    "name": "Tomato",
    "slug": "tomato",
  },
]
`;
